spring-batch-infrastructure/src/test/java/org/springframework/batch/item/data/MongoItemWriterTests.java

```diff
@@ -76,7 +78,7 @@ public void setUp() throws Exception {
 		when(this.template.bulkOps(any(), anyString())).thenReturn(this.bulkOperations);
 		when(this.template.bulkOps(any(), any(Class.class))).thenReturn(this.bulkOperations);
 
-		MappingContext mappingContext = new MongoMappingContext();
+		MappingContext<BasicMongoPersistentEntity<?>, MongoPersistentProperty> mappingContext = new MongoMappingContext();
```

Comment:
This is causing a compilation error. Changing it as follows fixes the issue and removes the raw type warning as well:

```
MappingContext<MongoPersistentEntity<?>, MongoPersistentProperty> mappingContext = new MongoMappingContext();
```

I will fix that on merge.